### PR TITLE
Image widgets: Centre everywhere except hints

### PR DESCRIPTION
**Pull request by jeremywiebe**
## Summary:

This PR updates image rendering so that `image` widgets (but _not_ bare Markdown image links) are rendered with new alignments:

  * In articles: centred
  * In exercises: centred
  * In hints: left

Note: The Image Alignment [1Pager](https://khanacademy.atlassian.net/wiki/spaces/LC/pages/2607218692/Allow+for+both+center-+and+left-aligned+images+in+articles+and+exercises+1P) requests more control than this PR provides so there may be some future work.  

However, LEMS-1798 lays out the requirements as I've stated them so this at least gets us to more aesthetically pleasing image rendering across the board. 

Issue: LEMS-1798

## Test plan:

Test on both Desktop and Mobile...

  * Navigate to the [Murals and Public Art in 1930s Rome](https://www.khanacademy.org/humanities/art-1010/german-art-between-the-wars/xdc974a79:nazi-fascist-visual-culture/a/murals-and-public-art-in-1930s-rome) article. Note that all images are either "full bleed" or centred horizontally if they are not the width of the article content. 

  * Navigate to the [Apply: ionic bonds and compounds](https://www.khanacademy.org/science/hs-chemistry/x2613d8165d88df5e:chemical-bonding/x2613d8165d88df5e:ionic-bonds/e/apply-ionic-bonds-and-compounds) exercise. Note that hints that contain an image are left-aligned (such as in the question "Strontium chloride is an ionic compound. It is used in fireworks to produce red light.").

  * Navigate to the [German art between the wars](https://www.khanacademy.org/humanities/art-1010/german-art-between-the-wars/test-german-art-between-wars/e/german-art-between-the-wars) exercise. Note that the images in the exercise content are centred.

## Examples from production

| device | url | screenshot |
| --- | --- | --- |
| Desktop | [German art between the wars](https://www.khanacademy.org/humanities/art-1010/german-art-between-the-wars/test-german-art-between-wars/e/german-art-between-the-wars) | Screenshot 2024-03-20 at 1 29 08 PM |
| Phone | [German art between the wars](https://www.khanacademy.org/humanities/art-1010/german-art-between-the-wars/test-german-art-between-wars/e/german-art-between-the-wars) | image | 
| Desktop | [Murals and Public Art in 1930s Rome](https://www.khanacademy.dev/humanities/art-1010/german-art-between-the-wars/xdc974a79:nazi-fascist-visual-culture/a/murals-and-public-art-in-1930s-rome) | Screenshot 2024-03-20 at 1 57 08 PM |
| Mobile | [Murals and Public Art in 1930s Rome](https://www.khanacademy.dev/humanities/art-1010/german-art-between-the-wars/xdc974a79:nazi-fascist-visual-culture/a/murals-and-public-art-in-1930s-rome) | image |
| Desktop | [Apply: ionic bonds and compounds](https://www.khanacademy.dev/science/hs-chemistry/x2613d8165d88df5e:chemical-bonding/x2613d8165d88df5e:ionic-bonds/e/apply-ionic-bonds-and-compounds) |  |
| Mobile | [Apply: ionic bonds and compounds](https://www.khanacademy.dev/science/hs-chemistry/x2613d8165d88df5e:chemical-bonding/x2613d8165d88df5e:ionic-bonds/e/apply-ionic-bonds-and-compounds) | image |

### CODE REVIEW

File: packages/perseus/src/styles/widgets/image.less [hunk 1]
```diff
@@ -52,3 +52,32 @@
         margin-top: 5px;
     }
 }
+
+// Image widgets outside of hints are always centred
+.perseus-image-widget:not(.perseus-hint-renderer *) {
```
Comment (jeremywiebe) [inline packages/perseus/src/styles/widgets/image.less:57]: This uses a Selectors Level 4 version of `:not()`. I had a hard time verifying if it was fully supported (caniuse.com was a bit ambiguous about browser support for the Level 4 version specifically). 

I tried on the latest of Safari, Chrome, and Firefox and it worked in all three. 🤷‍♂️

File: packages/perseus/src/__testdata__/server-item-renderer.testdata.ts [hunk 2]
```diff
@@ -359,3 +359,196 @@ export const itemWithLintingError: PerseusItem = {
     itemDataVersion: {major: 0, minor: 0},
     answer: null,
 };
+
+export const itemWithImages: PerseusItem = {
```
Comment (handeyeco) [inline packages/perseus/src/__testdata__/server-item-renderer.testdata.ts:363]: Consider looking at the `generateTestPerseusItem` helper

**Review by nishasy — APPROVED** [review]
The CSS makes sense to me.

I just think it would be neat if there were a storybook story (or stories) that's a more accurate representation of the contexts in which you'd see these images. The [current one](https://650db21c3f5d1b2f13c02952-rrcdwvviec.chromatic.com/?path=/story/perseus-renderers-server-item-renderer--image-examples-item) is in such a narrow container that it's hard to tell if the images are behaving how we want them to.

File: packages/perseus/src/__testdata__/server-item-renderer.testdata.ts [hunk 3]
```diff
@@ -359,3 +359,196 @@ export const itemWithLintingError: PerseusItem = {
...
+[[\u2603 image 2]]
+
+A small landscape image:
+[[\u2603 image 3]]
+
+A large portrait image:
+[[\u2603 image 4]]
+
+And what follows are _hints_...
+`,
```
Comment (handeyeco) [inline packages/perseus/src/__testdata__/server-item-renderer.testdata.ts:382]: **code style nit:** doing this kind of template stuff in an object just looks weird. I would recommend, in order of preference:
- using `"` and `\n` like existing test data
- making a separate variable to store this content so it's not in the object.

File: packages/perseus/src/styles/perseus-renderer.less [hunk 4]
```diff
@@ -252,7 +252,10 @@
     }
 
     .svg-image {
-        display: inline-block;
+        display: block;
+        margin: 0 auto;
+        margin-inline-start: auto;
+        margin-inline-end: auto;
```
Comment (handeyeco) [inline packages/perseus/src/styles/perseus-renderer.less:258]: Do we need this if this image is centered? If so, could we add a comment regarding what it's for?

File: packages/perseus/src/styles/widgets/image.less [hunk 5]
```diff
@@ -52,3 +52,24 @@
         margin-top: 5px;
     }
 }
+
+// Image widgets are centred, by default
```
Comment (handeyeco) [inline packages/perseus/src/styles/widgets/image.less:56]:
```suggestion
// Image widgets are centered, by default
```

File: packages/perseus/src/styles/widgets/image.less [hunk 6]
```diff
@@ -52,3 +52,24 @@
...
+.perseus-image-widget {
+    margin-inline-start: auto;
+    margin-inline-end: auto;
```
Comment (handeyeco) [inline packages/perseus/src/styles/widgets/image.less:59]: I think this is more confusing than it needs to be. I'm guessing it's for right-to-left? But we don't support RTL in any other capacity. I'm also not sure [it works](https://caniuse.com/?search=margin-inline-start) in all the [browsers we support](https://khanacademy.atlassian.net/wiki/spaces/ENG/pages/103612417/Browser+Support).

**Review by handeyeco — COMMENTED** [review]
Beautiful to see all this testing. I'm worried about browser support and clarity though.

**Pull request merged**